framework/contracts/native/ibc-client/src/commands.rs

```diff
@@ -206,32 +244,67 @@ pub fn execute_send_packet(
 
     let cfg = CONFIG.load(deps.storage)?;
 
-    // Verify that the sender is a proxy contract
-    let account_base = cfg
-        .version_control
-        .assert_proxy(&info.sender, &deps.querier)?;
+    // The packet we need to send depends on the action we want to execute
 
-    // get account_id
-    let account_id = account_base.account_id(deps.as_ref())?;
+    let note_message = match &action {
+        HostAction::Dispatch { .. } | HostAction::Helpers(_) => {
+            // Verify that the sender is a proxy contract
+            let account_base = cfg
+                .version_control
+                .assert_proxy(&info.sender, &deps.querier)?;
 
-    // Can only call non-internal actions
-    if let HostAction::Internal(_) = action {
-        return Err(IbcClientError::ForbiddenInternalCall {});
-    }
+            // get account_id
+            let account_id = account_base.account_id(deps.as_ref())?;
 
-    let callback_request = callback_info.map(|c| CallbackRequest {
-        receiver: env.contract.address.to_string(),
-        msg: to_json_binary(&IbcClientCallback::UserRemoteAction(c)).unwrap(),
-    });
+            let callback_request = callback_info.map(|c| CallbackRequest {
+                receiver: env.contract.address.to_string(),
+                msg: to_json_binary(&IbcClientCallback::UserRemoteAction(c)).unwrap(),
+            });
+
+            send_remote_host_action(
+                deps.as_ref(),
+                account_id,
+                account_base,
+                host_chain,
+                action,
+                callback_request,
+            )?
+        }
+        HostAction::Module {
+            msg,
+            source_module,
+            target_module,
+        } => {
+            // We make sure the source_module is indeed who they say they are
+            let registry = CONFIG
```

Comment:
This limits IBC module calls to modules that are exactly the same version. Is that the behaviour we want?

Reply:
That's not really what is happenning here. The source and target modules can be completely different and with mismatched versions

Reply:
In the current version, modules versions and considered only for address resolution